x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -13,8 +14,55 @@ import {
   dataAnalyticsExplainSchema,
   analyticsIdSchema,
   stopsDataFrameAnalyticsJobQuerySchema,
+  analyticsIdIndexSchema,
 } from './schemas/data_analytics_schema';
+import { IndexPatternHandler } from '../models/data_frame_analytics/index_patterns';
 
+function getIndexPatternId(context: RequestHandlerContext, patternName: string) {
+  const iph = new IndexPatternHandler(
+    context.ml!.mlClient.callAsCurrentUser,
+    context.core.savedObjects.client
+  );
+  return iph.getIndexPatternId(patternName);
+}
+
+function deleteIndexPatternById(context: RequestHandlerContext, indexPatternId: string) {
+  const iph = new IndexPatternHandler(
+    context.ml!.mlClient.callAsCurrentUser,
+    context.core.savedObjects.client
+  );
+  return iph.deleteIndexPatternById(indexPatternId);
+}
+
+async function userCanDeleteIndex(
+  context: RequestHandlerContext,
+  destinationIndex: string
+): Promise<boolean> {
+  let privilege;
+  try {
+    privilege = await context.ml!.mlClient.callAsCurrentUser('transport.request', {
+      method: 'POST',
+      path: '/_security/user/_has_privileges',
+      body: {
+        index: [
+          {
+            names: [destinationIndex], // uses wildcard
+            privileges: ['delete_index'],
+          },
+        ],
+      },
+      ignoreUnavailable: true, // we allow 404 incase the user shutdown security in-between the check and now
+    });
+    return privilege.index[destinationIndex].delete_index === true;
+  } catch (err) {
+    if (
+      err.message === 'no handler found for uri [/_security/user/_has_privileges] and method [POST]'
```

Comment:
the error message could change at any time. you might want to check for status code instead if it's possible

Reply:
Looking at `verifyHasPrivileges` from `x-pack/plugins/monitoring/server/lib/elasticsearch/verify_monitoring_auth.js` I think there's a particular reason why it's checked it like that. Let me follow up with the authors of that API.

Reply:
If we move to a check similar to our other privilege checks, this error message comparison won't be needed.
I believe the reason this text is checked it because `ignoreUnavailable` is set, so the error returned could be a 404 rather than just a privilege error.
Our other privilege checks first confirm that security is enabled before attempting a privilege check so this error message comparison isn't needed.
See my comment further down this PR.